### ISSUE TITLE
Add wear/uses to crafting replacements and chance to fail

### ISSUE THREAD

**Issue by MarkuBu** [minetest/minetest#5323]
It is possible to add replacements to crafting recipes to use tools or items in a recipe. But these items don't get wear and use them forever

I don't know if the 'on_use' callback is called if an item is used in a recipe. If not it would be helpful to add this to get some functionality like crafting failure

**Comment by paramat** [comment]
Author left MT, anyone else requesting this?

**Comment by rubenwardy** [comment]
It's worth noting this is possible with mods using `register_on_craft`, however it's arguable that it should be an API feature

**Comment by raymoo** [comment]
@rubenwardy 
`register_on_craft` is generally painful to use since to figure out the craft recipe that was used, you have to redo all the recipe matching work in Lua. Finding items also becomes annoying for recipes smaller than the grid.

**Comment by paramat** [comment]
Ok, don't understand the request though. Why would a recipe item need wear if it is used up iin a recipe?

**Comment by raymoo** [comment]
@paramat The idea is that it isn't used up, it just accumulates wear. You could for example have a saw item that you craft together with a log to get some wood planks. You keep the saw, but it gets some wear and eventually it will break and you will have to use a new saw.

**Comment by paramat** [comment]
Ok thanks, makes sense.

**Comment by jastevenson303** [comment]
The problem I am experiencing is an inability to set wear on anything other than a tool.  If that's the case, then that'd need to be addressed first before this issue here.

**Comment by raymoo** [comment]
@jastevenson303 I don't see why that needs to be addressed before this, they seem like orthogonal issues.

**Comment by jastevenson303** [comment]
Yeah, sorry, I'm sure I misunderstand the issue.  Is this issue about adding wear to "tools and items" once they've been used in a craft recipe, and return as replacements?

Because if that's the case then the replacement-with-wear would only work for tools, because non-tools return 0 for get_wear().

**Comment by raymoo** [comment]
@jastevenson303 Everything you've said is true, but even with only tools being able to have wear, this wear replacement feature would still be useful since you can have tools in recipes.

**Comment by jastevenson303** [comment]
Alright.  So if this issue is about adding wear to things once replaced, then my question is whether this is already possible.  Aren't there register_on_craft callbacks?

**Comment by Ferk** [comment]
@jastevenson303 for the engine a tool is preciselly an item with wear. If non-tool items (ie craftitems) were able to have wear there would be no sense to have any differenciation between tool and item.

Imho, the differenciation should either made more clear (and avoid having tools that do not have durability, like the keys), or be removed all together and deprecate "register_tool", imho.

This has already been discussed in the past: #5323 #4815 
There's also a PR to add wear to craftitems: #4715

**Comment by jastevenson303** [comment]
Thanks for the information, and education as to the different between craftitems and tools (I was wondering).

I just want the little green bar on a particular meshnode, which I cannot turn into a tool because then the wieldimage is broken.

**Comment by paramat** [comment]
Duplicates #1741
